lib/auth/token.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+/**
+ * Part of GDL gdl-frontend.
+ * Copyright (C) 2017 GDL
+ *
+ * See LICENSE
+ */
+
+import Cookie from 'js-cookie';
+import lscache from 'lscache';
+
+const ANON_ACCESS_TOKEN_KEY = 'anon-access-token';
+const AUTH_ACCESS_TOKEN_KEY = 'auth-access-token';
+export const LOGOUT_KEY = 'logout';
+
+/**
+ * Set anonymous tokens (for user not logged in)
+ */
+export function setAnonToken(token: {
+  access_token: string,
+  expires_in: number,
+}) {
+  if (!process.browser || token.access_token == null) {
+    return;
+  }
+
+  // The anon token has an expiration time (in seconds), so turn it into a datetime in the form of milliseconds
+  const expirationInMs = token.expires_in * 1000 + new Date().getTime();
+  // lscache uses minutes for expiration, while the token TTL is in seconds
+  lscache.set(ANON_ACCESS_TOKEN_KEY, token.access_token, Math.floor(token.expires_in / 60));
+  Cookie.set(ANON_ACCESS_TOKEN_KEY, token.access_token, {
+    expires: new Date(expirationInMs),
+  });
+}
+
+/**
+ * Set auth tokens (for user logging in)
+ */
+export function setAuthToken(token: { accessToken: string, expiresIn: number }) {
+  if (!process.browser || token.accessToken == null) {
+    return;
+  }
+  const expirationInMs = token.expiresIn * 1000 + new Date().getTime();
+  // lscache uses minutes for expiration, while the token TTL is in seconds
+  lscache.set(AUTH_ACCESS_TOKEN_KEY, token.accessToken, Math.floor(token.expiresIn / 60));
+  Cookie.set(AUTH_ACCESS_TOKEN_KEY, token.accessToken, {
+    expires: new Date(expirationInMs),
+  });
```

Comment:
The code above is duplicated in `setAnonToken`. Should it be factored out to avoid having to make (potential) changes two places?